### PR TITLE
Adding PKI Responses 3

### PR DESCRIPTION
**Pull request by AnPucel**
`path_fetch_*.go`

Please see this [RFC](https://github.com/hashicorp/vault/pull/18626) for implementation details.

| File | Path | Response |
| ------------- | ------------- | ------------- |
| `path_fetch` | `ca/pem`, `crl/*`,  `cert/*`| [Response](https://github.com/hashicorp/vault/blob/087ee94e7ba7300f8a50a099a7a698ab3c087434/builtin/logical/pki/path_fetch.go#L417) |
| `path_fetch_issuers` | `issuers/*` | List operation, should have `keys`|
| `path_fetch_issuers` | `issuer/:issuer_ref/*` | [Responses](https://github.com/hashicorp/vault/blob/087ee94e7ba7300f8a50a099a7a698ab3c087434/builtin/logical/pki/path_fetch_issuers.go#LL343C3-L343C3) |
| `path_fetch_keys` | `keys/*` | List operation, should have `keys` |
| `path_fetch_keys` | `key/*` | [Get](https://github.com/hashicorp/vault/blob/087ee94e7ba7300f8a50a099a7a698ab3c087434/builtin/logical/pki/path_fetch_keys.go#L229), [Update](https://github.com/hashicorp/vault/blob/087ee94e7ba7300f8a50a099a7a698ab3c087434/builtin/logical/pki/path_fetch_keys.go#L299)|

Related PRs:
- [PKI Config](https://github.com/hashicorp/vault/pull/18376)
- [PKI Issue & Manage](https://github.com/hashicorp/vault/pull/18479)
- [PKI Other](https://github.com/hashicorp/vault/pull/18612)

### CODE REVIEW

File: builtin/logical/pki/path_fetch_issuers.go [hunk 1]
```diff
@@ -20,6 +21,23 @@ func pathListIssuers(b *backend) *framework.Path {
...
+						Fields: map[string]*framework.FieldSchema{
+							"keys": {
+								Type:        framework.TypeStringSlice,
+								Description: `A list of keys`,
+								Required:    true,
+							},
+							"key_info": {
+								Type:        framework.TypeMap,
+								Description: `Key info with issuer name`,
+								Required:    true,
```
Comment (lursu) [inline builtin/logical/pki/path_fetch_issuers.go:36]:
```suggestion
								Required:    false,
```

Comment (lursu) [reply builtin/logical/pki/path_fetch_issuers.go:36]: Looks like this is similar to @danielehc found in my prs but this isn't set if keys is empty. https://github.com/hashicorp/vault/blob/3ce5bbb601efa6a4b7308b41475b49e8535b301f/sdk/logical/response.go#L159-L161

File: builtin/logical/pki/path_fetch_keys.go [hunk 2]
```diff
@@ -16,7 +17,24 @@ func pathListKeys(b *backend) *framework.Path {
...
+						Fields: map[string]*framework.FieldSchema{
+							"keys": {
+								Type:        framework.TypeStringSlice,
+								Description: `A list of keys`,
+								Required:    true,
+							},
+							"key_info": {
+								Type:        framework.TypeMap,
+								Description: `Key info with issuer name`,
+								Required:    true,
```
Comment (lursu) [inline builtin/logical/pki/path_fetch_keys.go:33]:
```suggestion
								Required:    false,
```

Comment (lursu) [reply builtin/logical/pki/path_fetch_keys.go:33]: same goes here. https://github.com/hashicorp/vault/blob/3ce5bbb601efa6a4b7308b41475b49e8535b301f/sdk/logical/response.go#L159-L161

**Review by lursu — APPROVED** [review]
LGTM just a couple of suggestions.

File: changelog/18596.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+openapi: Add openapi response definitions to PKI path_fetch_*.go
```
Comment (sgmiller) [inline changelog/18596.txt:2]: One comment is can we have a single changelog entry for all the response enhancements, rather than per file?

**Review by sgmiller — CHANGES_REQUESTED** [review]

File: changelog/18596.txt (hunk 3 above)
Comment (AnPucel) [reply changelog/18596.txt:2]: Sure thing! Thanks for pointing that out.

**Review by sgmiller — APPROVED** [review]

File: builtin/logical/pki/path_fetch.go [hunk 4]
```diff
@@ -382,6 +446,7 @@ func (b *backend) pathFetchRead(ctx context.Context, req *logical.Request, data
 
 reply:
 	switch {
+	// To Do: Should we add `http_raw_body` and `http_content_type` as optional in the response
```
Comment (averche) [inline builtin/logical/pki/path_fetch.go:449]:
```suggestion
```
I believe those are special parameters that will no show up in the final response.

**Review by averche — APPROVED** [review]

File: builtin/logical/pki/path_fetch.go (hunk 4 above)
Comment (AnPucel) [reply builtin/logical/pki/path_fetch.go:449]: yep. I think I added that a long time ago. Lemme remove it.

Comment (AnPucel) [reply builtin/logical/pki/path_fetch.go:449]: oh wait. I actually think these were breaking tests, which is why I addd them in the response

**Pull request merged**